AlertaDengue/dados/models.py

```diff
@@ -12,10 +12,21 @@ class ForecastModel(models.Model):
     commit_id    CHAR(7)    NOT    NULL,
     active    BOOL    NOT    NULL
     """
-    name = models.CharField(max_length=128, null=False)
-    total_weeks = models.IntegerField(null=False)
-    commit_id = models.CharField(max_length=7, null=False)
-    active = models.BooleanField(null=False)
+    name = models.CharField(
+        max_length=128, null=False,
+        help_text=_('Nome do Modelo de Previsão')
+    )
+    weeks = models.IntegerField(
+        null=False, help_text=_('Total de Semanas')
+    )
+    commit_id = models.CharField(
+        max_length=7, null=False,
+        help_text=_('ID do commit (github)')
+    )
+    active = models.BooleanField(
+        null=False,
+        help_text=_('Está ativo?')
+    )
 
     class Meta:
         db_table = 'Municipio\".\"forecast_model'
```

Comment:
Um erro está acontecendo ao acessar o admin para esse modelo, provavelmente por que falta uma migração.

Eu sei que o nosso uso atual do banco é confuso, e isso com certeza deveria melhorar, mas lembro que temos dois bancos: o banco `dengue` que guarda a maioria dos dados propriamente ditos e que é acessado por todas as aplicações envolvidas no infodengue (não só essa aplicação django aqui) e o banco `infodengue` que poderia ter um nome melhor, mas é o banco de dados gerido pelo django, que só é acessado por essa aplicação.

O django só gerencia o banco infodengue, o outro foi criado por fora, não recebe migrações dele etc. 

O que temos feito até agora é usar o banco infodengue e as convenções do django para gerenciar todas as tabelas referentes a modelos dessa aplicação. Por isso não mudamos em nenhum dos nossos outros modelos o nome da tabela com o atributo `db_table`. 

De qualquer jeito, eu não encontrei uma migração para esse modelo, então as tabelas não vão existir no banco.

Reply:
@flavioamieiro 
como esses dados estão relacionados com os dados do schema Municipio, da base de dados dengue, então tive que usar o db_table pois os dados não estarão no schema principal. 

outra forma de resolver isso, seria criar um usuário municipio e alterar o seu schema padrão para "Municipio".

conforme comentário (https://github.com/AlertaDengue/AlertaDengue/pull/143#issuecomment-334799014) é necessário executar o script presente no notebook Forecast.ipynb, pois além de criar as tabelas, será criado dados de exemplo para testar a visualização.

em produção somente é necessário executar o script: https://github.com/AlertaDengue/AlertaDengue/blob/develop/sql/create_table_forecast.sql

Reply:
@xmnlab eu acho que não entendi muito bem: o forecast tem que estar no banco `dengue`? Eu rodei o script `create_table_forecast.sql` no banco `dengue`. Eu deveria rodar ele no banco `infodengue`?

Se o forecast tem que estar no `dengue`, acho que não vai funcionar (pelo menos não com a estrutura do código que temos agora) por que o model está tentando acessar o banco `infodengue`. Pelo menos a página do admin tenta usar o model direto, então ele tenta fazer a query no banco `infodengue`.

O django só conhece o banco 'infodengue`. É por isso que o arquivo [dbdata.py](https://github.com/AlertaDengue/AlertaDengue/blob/develop/AlertaDengue/dados/dbdata.py), por exemplo, precisa definir a própria conexão ao banco de dados pra pegar dados do banco `dengue`, já que a conexão normal do django ao banco não chega lá.

Me desculpa pela confusão, mas eu acho que não entendi direito as mudanças mesmo.

Reply:
@flavioamieiro seria no banco dengue.

É estranho, porque aqui no meu ambiente funcionou rsrs acho que agora fiquei confuso também.

vou fazer novamente o restore e refazer os testes aqui.

Reply:
@xmnlab Ok. No ambiente de staging eu rodei o script no banco `dengue`, então acho que isso está certo. O que eu não fiz foi rodar o notebook. Posso até fazer isso, mas acho melhor a gente encontrar o erro antes, por que o admin tem que ser capaz de abrir a página sem dados de teste.

Tô tentando também aqui encontrar o problema.

Reply:
Só como referência, tentei criar a migração inicial para a app dados e encontrei o seguinte erro:

```
$ python manage.py makemigrations dados
Migrations for 'dados':
  dados/migrations/0001_initial.py
    - Create model ForecastCity
    - Create model ForecastModel
    - Add field forecast_model to forecastcity
    - Alter unique_together for forecastcity (1 constraint(s))
(infodengue) [amieiro@stallman ~/.virtualenvs/infodengue/src/AlertaDengue(PR-143)] $ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dados, dbf, sessions
Running migrations:
  Applying dados.0001_initial...Traceback (most recent call last):
  File "/home/amieiro/.virtualenvs/infodengue/lib/python3.5/site-packages/django/db/backends/utils.py", line 63, in execute
    return self.cursor.execute(sql)
psycopg2.ProgrammingError: schema "Municipio" does not exist
LINE 1: CREATE TABLE "Municipio"."forecast_city" ("id" serial NOT NU...
```

Ou seja: ele está mesmo tentando acessar o banco `infodengue` para criar essa tabela (e como o schema "Municipio" não existe nesse banco, ele encontra um erro. Acho que exatamente a mesma coisa está acontecendo no admin quando acessamos a url `/admin/dados/forecastcity/`: ele tenta acessar a tabela "Municipio"."forecast_city" no banco `infodengue` e não consegue. Eu não conheço um jeito simples de fazer o admin usar um banco externo à aplicação.

Reply:
pode ser então alguma questão de configuração de ambiente. 

o sql de criação das tabelas estava desatualizado, tinha que trocar de geoid para geocode.

poderias enviar-me por email seus arquivos de configuração (settings.ini, local_settings, etc)? 

uma dúvida: as bases precisam estar em bancos separados? não resolveria apenas separar em schemas diferentes?

Reply:
talvez de pra implementar isso: https://docs.djangoproject.com/en/1.11/topics/db/multi-db/
mas antes precisava dos seus arquivos de configuração para poder criar um ambiente igual ao teu.

Reply:
@flavioamieiro consegui replicar aqui.

eu tinha um arquivo local_settings, onde eu forçava o dbname:

```python

from AlertaDengue.settings import *

INSTALLED_APPS += ('django_extensions',)


DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': PSQL_DB,
        'USER': PSQL_USER,
        'PASSWORD': PSQL_PASSWORD,
        'HOST': PSQL_HOST,
        'PORT': '',
    }
}
```

Reply:
@flavioamieiro adicionei um router para que cada db possa usar uma db distinta

Reply:
@xmnlab aqui vai o .env que eu uso pra desenvolvimento (quando rodo no docker é um pouco diferente, mas nada que deveria ter impacto, só configurações específicas do docker):

```
ALLOWED_HOSTS=alerta.dengue.mat.br,info.dengue.mat.br,localhost
SECRET_KEY=my-secret-key
DEBUG=True
PSQL_PASSWORD=XXXX
DATABASE_URL=postgres://dengueadmin:XXXX@localhost:5432/infodengue
EMAIL_BACKEND=django.core.mail.backends.console.EmailBackend
PSQL_HOST=127.0.0.1
QUERY_CACHE_TIMEOUT=0
```

Sobre a necessidade dos bancos serem diferentes: a gente mantém isso principalmente por razões históricas e por que várias outras aplicações mexem no banco `dengue`. Esse segundo motivo é o principal, e é também a razão pra evitarmos escrever nesse banco sempre que possível. Usamos ele só pra leitura em quase todos os lugares. A exceção é o load_sinan, que precisa escrever no banco.

Eu gosto da ideia de usarmos o multi_db, mas ainda acho importante mantermos o cuidado com esse banco compartilhado. Não acho que teremos problemas com o admin escrevendo lá, mas é só algo pra se manter em mente.

Reply:
@flavioamieiro obrigado!

podemos alterar o settings.py para que o DATABASE_URL utiliza parametros do settings.ini?

acho que sería mais claro que essa variavel deva ser configurada.

Reply:
@flavioamieiro 

desculpa .. agora que percebi .. já tem .. é que não era muito claro: 

```
DATABASES = {
    'default': config('DATABASE_URL', default='sqlite:///geodjango.db', cast=db_url)
}
```